frappe/public/js/frappe/views/reports/report_view.js

```diff
@@ -66,6 +66,131 @@ frappe.views.ReportView = class ReportView extends frappe.views.ListView {
 			args: this.order_by,
 			onchange: this.on_sort_change.bind(this)
 		});
+
+		//Setup groupby for reports
+		this.setup_groupby_area();
+	}
+
+	make_groupby_button() {
+		let group_by_button =  $(`<div class="tag-groupby-area">
+								<div class="active-tag-groupby">
+									<button class="btn btn-default btn-xs add-groupby text-muted">
+											${__("Add Group")}
+									</button>
+								</div>
+							</div>`);
+		this.page.wrapper.find(".sort-selector").before(group_by_button);
+		group_by_button.click(() => this.groupby_edit_area.show());
+	}
+
+	setup_groupby_area() {
```

Comment:
lets make a new class for the GroupBy control, that we can move to another file.

Also add code comments.